From cb2ead4b9fc17554a8694fd213bfb100d9c15678 Mon Sep 17 00:00:00 2001
From: Gerd Hoffmann <kraxel@redhat.com>
Date: Fri, 20 Jan 2023 12:59:33 +0100
Subject: [PATCH] 90-uki-copy.install: create $BOOT/EFI/Linux directory if
 needed

Do not consider a missing 'Linux' subdirectory an error.
Just create it instead.

(cherry picked from commit c7314ee7e290b3978e2f2d7726d07656eda071f9)

Related: RHEL-16354
---
 src/kernel-install/90-uki-copy.install | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/kernel-install/90-uki-copy.install b/src/kernel-install/90-uki-copy.install
index d6e3deb723..6c71b211d7 100755
--- a/src/kernel-install/90-uki-copy.install
+++ b/src/kernel-install/90-uki-copy.install
@@ -49,8 +49,8 @@ case "$COMMAND" in
 esac
 
 if ! [ -d "$UKI_DIR" ]; then
-    echo "Error: entry directory '$UKI_DIR' does not exist" >&2
-    exit 1
+    [ "$KERNEL_INSTALL_VERBOSE" -gt 0 ] && echo "creating $UKI_DIR"
+    mkdir -p "$UKI_DIR"
 fi
 
 TRIES_FILE="${KERNEL_INSTALL_CONF_ROOT:-/etc/kernel}/tries"
